frb_codegen/src/generator/dart/mod.rs

```diff
@@ -104,12 +110,20 @@ fn get_dart_api_spec_from_ir_file(ir_file: &IrFile, block_index: BlockIndex) ->
     let dart_funcs = ir_file
         .funcs
         .iter()
-        .map(generate_api_func)
+        .map(|f| generate_api_func(f, ir_file))
         .collect::<Vec<_>>();
     let dart_structs = distinct_types
         .iter()
-        .map(|ty| TypeDartGenerator::new(ty.clone(), ir_file).structs())
+        .map(|ty| {
+            {
+                let mut t = TypeDartGenerator::new(ty.clone(), ir_file);
+                t.set_dart_api_class_name(dart_api_class_name);
```

Comment:
As discussed here https://github.com/fzyzcjy/flutter_rust_bridge/pull/543#issuecomment-1172990930, maybe we can avoid mutability?